nbdime/diff_format.py

```diff
@@ -293,57 +293,76 @@ def _accum(seq):
             yield total
 
 
+_addops = (DiffOp.ADD, DiffOp.ADDRANGE)
+
+
 def _check_overlaps(existing, new):
     """Check whether existing collection of diff ops shares a key with the
     new diffop, and if they  also have the same op type.
     Assumes exsiting diff ops are sorted on key.
     """
     for oo in reversed(existing):
-        if oo.key == new.key and oo.op == new.op:
-            # Found a match, combine ops
-            return oo
+        if oo.key == new.key:
+            if oo.op == new.op:
+                # Found a match, combine ops
+                return oo
+            elif oo.op in _addops and new.op in _addops:
+                # Addrange and single add can both point to same key
+                return oo
+
 
 def _combine_ops(existing, new):
     """Combines new op into existing op
     """
-    if new.op == DiffOp.ADDRANGE:
-        existing.valuelist += new.valuelist
+    if new.op in _addops:
+        if existing.op == DiffOp.ADD:
+            existing.op == DiffOp.ADDRANGE
+            existing.valuelist = [existing.value]
+            del existing.value
```

Comment:
`del` won't work on `DiffEntry`, I'd rewrite the code to consider existing immutable and recreate a new diffentry object instead.